### PR TITLE
Use 1 custom URL manager for Twitch clips

### LINKED ISSUE #171: Merge 2 resolver.CustomURLManager objects into 1 in twitch's custom resolver
Currently we declare two separate custom resolvers for twitch clips Which is inefficient.
This can be easily avoided by applying simiar approach with regular expression like in Discord resolver or a domain map in BetterTTV one.
I could try fixing this on my own to learn something new peepoHackies

Code I'm talking about:
https://github.com/Chatterino/api/blob/5bc8059a719035d7169d2b127aad19c234c2fa5c/internal/resolvers/twitch/resolver.go#L74-L106

### PR DESCRIPTION
**Pull request by zneix**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

~~Uses regex (just like Discord invite links) instead of domain map, because it's simply easier.~~
Uses complex check and run methods. Added some check tests to make review process and understanding check function easier.

Closes #171

### CODE REVIEW

**Review by pajlada — CHANGES_REQUESTED** [review]
This regex amalgamation is not an improvement - try to find another solution

File: internal/resolvers/twitch/check_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package twitch
```
Comment (pajlada) [inline internal/resolvers/twitch/check_test.go:1]: good tests

File: internal/resolvers/twitch/check.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import (
+	"net/url"
+	"regexp"
+
+	"github.com/Chatterino/api/pkg/utils"
+)
+
+var (
+	clipSlugRegex = regexp.MustCompile(`^\/(\w{2,25}\/clip\/)?([a-zA-Z0-9]+(?:-[-\w]{16})?)$`)
```
Comment (pajlada) [inline internal/resolvers/twitch/check.go:11]: Do we have documentation to confirm we can be this strict with the "extra part" behind the clip slug? I'm referring mostly to the fact that we only consider it valid if it's 16 characters

Comment (zneix) [reply internal/resolvers/twitch/check.go:11]: We don't. This regex is just based on many examples and tests that we've done with Supinic and Leppunen when they were updating their smart things to new format of clips.

Comment (zneix) [reply internal/resolvers/twitch/check.go:11]: I understand that in the future Twitch might change their clip slug format once again - possibly to make the "extra part" longer than 16 characters, but I think we're good for now, as this new format allows for loads of new clips to be generated, so the probability of this occuring is very low.

Comment (pajlada) [reply internal/resolvers/twitch/check.go:11]: Would we hit any harmful false positives if we just let it be anything about X amount of characters? i.e. 8 characters

Comment (zneix) [reply internal/resolvers/twitch/check.go:11]: I guess nothing bad would happen. If anything, we'd be just less strict about what we pass to load function, but realistically speaking that's no problem at all.
I'm guessing you're willing to change this `{16}` to `+`. If so feel free to make that change and merge the PR.

Comment (pajlada) [reply internal/resolvers/twitch/check.go:11]: Fair enough - I guess it's more of a "do effort now" vs "do effort then" question, but it's not even sure us changing it to a `+` to solve it so I'll approve it as is

**Review by pajlada — APPROVED** [review]
:+1: Free to merge whenever

**Pull request merged**